File: account/src/main/java/serp/project/account/core/domain/dto/request/CreateModuleDto.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.account.core.domain.dto.request;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
public class CreateModuleDto {
    @NotBlank(message = "Name cannot be blank")
    private String name;
    @NotBlank(message = "Code cannot be blank")
    private String code;
    private String description;
    @NotBlank(message = "Keycloak Client ID cannot be blank")
    private String keycloakClientId;
    private String category;
    private String icon;
    private Integer displayOrder;
    private String moduleType;
    private Boolean isGlobal;
    private Long organizationId;
    private Boolean isFree;
    private String pricingModel;
    private String status;
}

File: api_gateway/src/core/domain/dto/request/account/create_module_dto.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package request

type CreateModuleDto struct {
	Name             string  `json:"name"`
	Description      *string `json:"description"`
	KeycloakClientId string  `json:"keycloakClientId"`
	Category         *string `json:"category"`
	Icon             *string `json:"icon"`
	DisplayOrder     *int    `json:"displayOrder"`
	ModuleType       *string `json:"moduleType"`
	IsGlobal         *bool   `json:"isGlobal"`
	OrganizationId   *int64  `json:"organizationId"`
	IsFree           *bool   `json:"isFree"`
	PricingModel     *string `json:"pricingModel"`
	Status           *string `json:"status"`
}
